python/ray/tune/examples/lightgbm_example.py

```diff
@@ -1,39 +1,68 @@
 import lightgbm as lgb
-import numpy as np
 import sklearn.datasets
 import sklearn.metrics
 from sklearn.model_selection import train_test_split
 
 from ray import tune
 from ray.tune.schedulers import ASHAScheduler
-from ray.tune.integration.lightgbm import TuneReportCheckpointCallback
+from ray.tune.integration.lightgbm import (
+    TuneReportCheckpointCallback,
+    TuneReportCallback,
+)
 
 
 def train_breast_cancer(config):
+    # This is a simple training function to be passed into Tune
+    # Load dataset
     data, target = sklearn.datasets.load_breast_cancer(return_X_y=True)
-    train_x, test_x, train_y, test_y = train_test_split(data, target, test_size=0.25)
-    train_set = lgb.Dataset(train_x, label=train_y)
-    test_set = lgb.Dataset(test_x, label=test_y)
-    gbm = lgb.train(
-        config,
-        train_set,
-        valid_sets=[test_set],
-        valid_names=["eval"],
-        verbose_eval=False,
-        callbacks=[
-            TuneReportCheckpointCallback(
-                {
-                    "binary_error": "eval-binary_error",
-                    "binary_logloss": "eval-binary_logloss",
-                }
-            )
-        ],
-    )
-    preds = gbm.predict(test_x)
-    pred_labels = np.rint(preds)
-    tune.report(
-        mean_accuracy=sklearn.metrics.accuracy_score(test_y, pred_labels), done=True
-    )
+
+    if config.pop("use_cv", False):
+        train_set = lgb.Dataset(data, label=target)
+
+        # Run CV, using the Tune callback
+        lgb.cv(
+            config,
+            train_set,
+            verbose_eval=False,
+            stratified=True,
+            # Checkpointing is not supported for CV
+            callbacks=[
+                TuneReportCallback(
+                    {
+                        "binary_error": "cv_agg-binary_error-mean",
```

Comment:
Are these aggregations done by `lgb.cv` automatically?

Reply:
Naive question, why is xgboost example different?

Reply:
Yes, lightgbm aggregates automatically

Reply:
It may be worthwhile to mention it somewhere. It's not entirely obvious to me.